jekyll/_cci2/build-publish-snap-packages.md

```diff
@@ -0,0 +1,177 @@
+base64 snapcraft.login | xsel --clipboard
+```
+
+1. Create a Snapcraft "login file" on your local machine that we upload to CircleCI. Assuming your local machine already has these tools installed and you are logged in to the Snapcraft Store (`snapcraft login`), we use the command `snapcraft export-login snapcraft.login` to generate a login file called `snapcraft.login`. As we don't want this file visible to the public or stored in the Git repository, we will base64 encode this file and store it in a [private environment variable][private-envars] called `$SNAPCRAFT_LOGIN_FILE`.
+
+```YAML
+...
+      - run:
+          name: "Publish to Store"
+          command: |
+            mkdir .snapcraft
+            echo $SNAPCRAFT_LOGIN_FILE | base64 --decode --ignore-garbage > .snapcraft/snapcraft.cfg
+            snapcraft push *.snap --release stable
+...
+```
+
+2. Once the base64 encoded version of the file is stored on CircleCI as a private environment variable, we can then use it within a build to automatically publish to the store.
+
+In this example, Snapcraft automatically looks for login credentials in `.snapcraft/snapcraft.cfg` so we decode the environment variable we made previously right into that location. `snapcraft push` is then used to upload the .snap file into the Snap Store.
+
+### Uploading vs Releasing
+
+`snapcraft push *.snap` by default will upload the snap to the Snap Store, run any store checks on the server side, and then stop. The snap won't be "released" meaning users won't automatically see the update. The snap can be published locally with the `snap release <release-id>` command or by logging into the Snap Store and clicking the release button.
+
+In typical CircleCI fashion, we can go fully automated (as in the above example) but using the `--release <channel>` flag. This uploads the snap, does Store side verification, and then will automatically release the snap in the specified channels. 
+
+
+## Workflows
+
+We can utilize multiple jobs to better organize our snap build. A job to build/compile the actual project, a job to build the snap itself, and a job that published the snap (and other packages) only on `master` would all be useful.
+
+[Workflows][workflows] can help with building snaps in two ways:
+
+1. **Snap Store Channels** - As we mentioned in the previous section, when we upload to the Store we could optionally release at the same time. This allows us to designate specific jobs on CircleCI to deploy to specific Snap Channels. For example, the `master` branch could be used to deploy to the `edge` channel` while tagged releases could be used to deploy to the `stable` channel.
+1. **Parallelize Packing** - If your software is being packaged as a snap as well as something else, say a flatpak, .deb, .apk, etc, each package type could be placed in its own job and all run parallel. This allows your build to complete must fast than if the .deb package could start to build until the snap completed, and so on.
```

Comment:
.apk-> or .apk
etc -> for example
This allows your build to complete must fast than if the .deb package could start to build until the snap completed, and so on. -> This allows your build to complete faster because the .deb package job doesn't have to wait until the snap package job completes.